services/ml/execution_impl_ie.cc

```diff
@@ -31,21 +42,52 @@ int32_t ExecutionImplIe::Init(int32_t preference) {
       device_name = "GPU";
     } else if (preference == mojom::PREFER_LOW_POWER) {
       device_name = "MYRIAD";
+    } else if (preference == mojom::PREFER_ULTRA_LOW_POWER) {
+      device_name = "GNA";
+      // Release in squence to avoid crash. Close GNA device befere re-open,
+      s_gna_infer_request.reset(nullptr);
+      s_gna_execution.reset(nullptr);
+      s_gna_plugin.reset(nullptr);
     }
     DLOG(INFO) << "[IE] Trying to get plugin by device name " << device_name;
-    plugin_.reset(
+    std::unique_ptr<InferenceEngine::InferRequest> infer_request;
+    std::unique_ptr<InferenceEngine::InferencePlugin> plugin;
+    std::unique_ptr<InferenceEngine::ExecutableNetwork> execution;
+    plugin.reset(
         new ie::InferencePlugin(static_cast<ie::InferenceEnginePluginPtr>(
-            ie::PluginDispatcher({""}).getPluginByDevice(device_name))));
-    const ie::Version* version = plugin_->GetVersion();
+            ie::PluginDispatcher({
+#if defined(OS_WIN)
+              L""  // Windows support UNICODE.
+#else
+              ""
+#endif
+            }).getPluginByDevice(device_name))));
+    const ie::Version* version = plugin->GetVersion();
     DLOG(INFO) << "[IE] succeed to load plugin " << version->buildNumber << " "
                << version->description;
-    execution_.reset(
+    std::map<std::string, std::string> plugin_Config = {};
+    if (preference == mojom::PREFER_ULTRA_LOW_POWER && input_scale_ > 0) {
+      std::string scaleFactorConfigKey = GNA_CONFIG_KEY(SCALE_FACTOR);
+      plugin_Config[scaleFactorConfigKey] = std::to_string(input_scale_);
```

Comment:
can you explain what the usage of setting `SCALE_FACTOR` by `input_scale`?

Reply:
the GNA Inference Engine plugin quantizes the model and input feature vector sequence to integer representation before performing inference. so the scale factor for input feature vector need to be set to GNA plugin, otherwise we need to quant the input data in js and pass the quantized data to WebNN.

Reply:
Is GNA model using quant8 or float32? Usually only quant8 data types use `scale`.

Reply:
Model is float32 including input data/weights/biases,but GNA plugin inference use int8/int16, so the model need to be quantized with the SCALE_FACTOR.

Reply:
I see. Where does the scale factor come from? The existing way to support quant8 model is to quantize the model offline and load the quantized model to WebNN API by suing `TENSOR_QUANT8_ASYMM` or `TENSOR_QUANT8_ASYMM_SIGNED` data types.

Reply:
It's calculated from the first utterance in the input ARK data seeing [speech_sample L.601](https://github.com/opencv/dldt/blob/2020/inference-engine/samples/speech_sample/main.cpp#L601). GNA plugin can quantize model when loading network seeing [gna_plugin inplementation](https://github.com/opencv/dldt/blob/2018/inference-engine/src/gna_plugin/gna_plugin.cpp#L1502). Do you mean we should use offline quantized model not float32 model?

Reply:
Thanks for the explanation. The existing WebNN API doesn't support to infer float32 model with int8 precision. So I am fine with the current solution as a workaround. Could you please file an issue about this workaround? For example, the title could be "Fix the workaround of using scale of float32 model to do int8 inference on GNA plugin".

Reply:
Done with filing [the issue](https://github.com/intel/webml-polyfill/issues/1191), Thanks.

Reply:
Looks good. Thanks!